rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/FileDownloadRequestChain.java

```diff
@@ -103,17 +105,20 @@ public Long call() throws IOException
     File file = new File(localFile);
     if (!file.exists()) {
       log.debug("Creating localfile : " + localFile);
-      String metadataFilePath = CacheUtil.getMetadataFilePath(remotePath, conf);
-      File mdFile = new File(metadataFilePath);
-      if (mdFile.exists() && mdFile.length() > 0) {
-        // Making sure when a new file gets created, we invalidate the existing metadata file
-        bookKeeper.invalidateFileMetadata(remotePath);
-      }
       file.setWritable(true, false);
       file.setReadable(true, false);
       file.createNewFile();
     }
 
+    long highestReadRequestLength = readRequests
+            .stream()
+            .map(readRequest -> readRequest.getActualReadEnd() - readRequest.getActualReadStart())
+            .max(Long::compareTo)
+            .get();
+    int remoteReadBufferSize = Math.min(maxRemoteReadBufferSize,
+            Math.toIntExact(Math.min(Integer.MAX_VALUE, highestReadRequestLength)));
```

Comment:
Using readRequest.getActualReadLength would cast to int and this part should simplify

Reply:
Same here, `readRequest.getActualReadEnd() - readRequest.getActualReadStart()` could be huge  that can cause int overflow. All this is to avoid int overflows